Milestone draft (#286)
finos/common-cloud-controls: docs/governance/community-guidelines/releases.md

# Releases

> This document is a [community guideline].

Reviewer: Links are giving me a 404 but that might just be the GitHub preview thing

Author: Works for me locally (i.e. it opens the correct page in VS code when clicking on it) so I presume it is a preview thing.

Reviewer: Yeah - link-checker is still showing the same amount of fails as before so should be good 👍

## Purpose

> The document outlines the guidelines of managing releases within the CCC project.

Reviewer: Should probably use "Common Cloud Controls (CCC)" the first time and the acronym for all the others

## Ownership

> The [Communications WG] takes ownership of releases. They set the roadmap and communicate project progress to the wider community. All other [WG]s should contribute to determining the content of each release.

Reviewer: ```suggestion
The [Communications WG] has overall ownership of releases. They are responsible for setting the roadmap and communicating project progress to the wider community. All other [WG]s should contribute to determining the content of a release.
```

## Scheduling

> The community aims for several releases per year. Releases can occur as soon as the controls for a service are complete and approved. The CCC project should prioritize stable releases and where possible avoid excessive release frequency (e.g., multiple times per day).

Reviewer: There's quite a range between "several a year" vs "multiple per day". More an observation than anything else

Reviewer: What constitutes stable in this context?

Author: not multiple releases per day :)

### Events

> The community can align releases with events such as the Open Source in Finance Forum (OSFF). These events provide excellent opportunities to showcase the project and generating interest.

Reviewer: worth adding a link for OSFF?

## Project Board Management

To manage GitHub issues targeted for a release, we recommend using a combination of [Milestones] and labels.

### Milestones

[Milestones] group GitHub issues related to a specific release. Each release should have a single top-level milestone associated with all relevant issues.

There are two types of releases:

1. Service Releases: Controls for individual services can be released independently.
2. Service Sets: When appropriate, the controls of multiple services can be grouped together and released as a bundle.

### Labels

Each [WG] should apply their label to relevant issues, clearly indicating which issues fall under their responsibility.

Additionally, labels can flag issues specifically targeted for events (e.g., OSFF events).

[WG]: <../community-groups.md#working-groups>
[Communications WG]: <../working-groups/communications/charter.md>

> [FINOS Point of Contact]: <../finos-poc.md>

Reviewer 2: Can we remove this if it's not going to be used?
